src/main/java/org/killbill/billing/plugin/adyen/api/AdyenPaymentPluginApi.java

```diff
@@ -416,35 +425,34 @@ private PaymentTransactionInfoPlugin executeFollowUpTransaction(final Transactio
         final DateTime utcNow = clock.getUTCNow();
 
         final PaymentModificationResponse response;
-        try {
-            response = transactionExecutor.execute(transactionAmount, paymentProvider, pspReference, splitSettlementData);
-        } catch (final ModificationFailedException e) {
-            return new AdyenPaymentTransactionInfoPlugin(kbPaymentId, kbTransactionId, transactionType, amount, currency, PaymentServiceProviderResult.ERROR, utcNow, e);
+        response = transactionExecutor.execute(transactionAmount, paymentProvider, pspReference, splitSettlementData);
+        if (!response.isSuccess()) {
+            return new AdyenPaymentTransactionInfoPlugin(kbPaymentId, kbTransactionId, transactionType, amount, currency, Optional.<PaymentServiceProviderResult>absent(), utcNow, response);
```

Comment:
Traditionally (i.e. in other plugins), the response table would contain entries for all calls, even failed ones (e.g. socket exceptions). That's quite useful for debugging. If it's possible, we should do the same here (unclear though if it impacts the logic when we read from the table, if it does, we can tackle that in another PR).